service/src/test/java/uk/gov/hmcts/reform/fpl/service/ConfirmApplicationReviewedServiceTest.java

```diff
@@ -0,0 +1,150 @@
+            .confirmApplicationReviewedEventData(ConfirmApplicationReviewedEventData.builder()
+                .confirmApplicationReviewedList(asDynamicList(List.of(NEW_BUNDLE_1, NEW_BUNDLE_2),
+                    NEW_BUNDLE_1.getId(), AdditionalApplicationsBundle::toLabel))
+                .build())
+            .build();
+
+        Element<AdditionalApplicationsBundle> result =
+            confirmApplicationReviewedService.getSelectedApplicationsToBeReviewed(caseData);
+
+        assertThat(result).isEqualTo(NEW_BUNDLE_1);
+    }
+
+    @Test
+    void shouldThrowExceptionWhenSelectedBundleNotFound() {
+        CaseData caseData = CaseData.builder()
+            .additionalApplicationsBundle(List.of(REVIEWED_BUNDLE, NEW_BUNDLE_1, NEW_BUNDLE_2))
+            .confirmApplicationReviewedEventData(ConfirmApplicationReviewedEventData.builder()
+                .confirmApplicationReviewedList(asDynamicList(List.of(NEW_BUNDLE_1, NEW_BUNDLE_2),
+                    UUID.randomUUID(), AdditionalApplicationsBundle::toLabel))
+                .build())
+            .build();
+
+        assertThatThrownBy(() -> confirmApplicationReviewedService.getSelectedApplicationsToBeReviewed(caseData));
+    }
+
+    @Test
+    void shouldMarkSelectedApplicationsAsReviewed() {
+        CaseData caseData = CaseData.builder()
+            .additionalApplicationsBundle(List.of(REVIEWED_BUNDLE, NEW_BUNDLE_1, NEW_BUNDLE_2))
+            .confirmApplicationReviewedEventData(ConfirmApplicationReviewedEventData.builder()
+                .confirmApplicationReviewedList(asDynamicList(List.of(NEW_BUNDLE_1, NEW_BUNDLE_2),
+                    NEW_BUNDLE_1.getId(), AdditionalApplicationsBundle::toLabel))
+                .build())
+            .build();
+
+        List<Element<AdditionalApplicationsBundle>> resultList =
+            confirmApplicationReviewedService.markSelectedBundleAsReviewed(caseData);
+
+        List<Element<AdditionalApplicationsBundle>> expectedList =
+            List.of(REVIEWED_BUNDLE,
+                element(NEW_BUNDLE_1.getId(), NEW_BUNDLE_1.getValue().toBuilder().applicationReviewed(YES).build()),
+                NEW_BUNDLE_2);
+
+        assertThat(resultList).isEqualTo(expectedList);
+    }
+
+    @Test
+    void shouldThrowExceptionSelectedApplicationsCannotBeMarkedAsReviewed() {
+        CaseData caseData = CaseData.builder()
+            .additionalApplicationsBundle(List.of(REVIEWED_BUNDLE, NEW_BUNDLE_1, NEW_BUNDLE_2))
+            .confirmApplicationReviewedEventData(ConfirmApplicationReviewedEventData.builder()
+                .confirmApplicationReviewedList(asDynamicList(List.of(NEW_BUNDLE_1, NEW_BUNDLE_2),
+                    UUID.randomUUID(), AdditionalApplicationsBundle::toLabel))
+                .build())
+            .build();
+
+        assertThatThrownBy(() -> confirmApplicationReviewedService.markSelectedBundleAsReviewed(caseData));
+    }
```

Comment:
Should test what is the exception thrown? Like
```
        assertThatThrownBy(() -> configuration.getCafcass(null))
            .isInstanceOf(NullPointerException.class)
            .hasMessage("Local authority code cannot be null");
```

Reply:
updated